internal/controller/utils.go

```diff
@@ -149,6 +149,17 @@ func createOrPatchStatefulSet(ctx context.Context, logger logr.Logger, ec *ecv1a
 		},
 	}
 
+	if len(ec.Spec.EtcdOptions) > 0 {
+		for i := 0; i < len(podSpec.Containers); i++ {
+			if podSpec.Containers[0].Name == "etcd" {
```

Comment:
Do you expect that `etcd` container will always be the first one, index 0?

Reply:
Good pick up.

This was meant to handle a scenario where the etcd container isn't at index 0, instead I botched it. Will push an update shortly

Reply:
Updated loop to fix issue